pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -356,16 +348,21 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 					continue
 				}
 
-				for _, cn := range con.Clusters {
+				previous := sets.NewSet(con.Clusters...)
+				current := sets.NewSet(clusters...)
+
+				// removed clusters
+				for cn := range previous.Difference(current) {
 					s.removeEdsCon(cn, con.ConID)
 				}
 
-				for _, cn := range clusters {
-					s.getOrAddEdsCluster(cn, con.ConID, con)
+				// new added clusters
+				for cn := range current.Difference(previous) {
+					s.getOrAddEdsCluster(cn, con)
 				}
 
 				con.Clusters = clusters
-				adsLog.Debugf("ADS:EDS: REQ %s %s clusters:%d", peerAddr, con.ConID, len(con.Clusters))
+				adsLog.Infof("ADS:EDS: REQ %s %s clusters:%d", peerAddr, con.ConID, len(con.Clusters))
```

Comment:
way to verbose, can we keep at info?

Reply:
+1. In general better to keep these messages at Info

Reply:
ok